utils/mvn/utils.go

```diff
@@ -48,48 +50,91 @@ func RunMvn(configPath, deployableArtifactsFile string, buildConf *utils.BuildCo
+	if err != nil {
+		return "", err
+	}
+	// Finding the relevant "Maven home" line in command response.
+	for _, line := range output {
+		if mavenHome == "" && strings.HasPrefix(line, "Maven home:") {
+			mavenHome, err = parseMvnHome(line)
+			if err != nil {
+				return "", err
 			}
 		}
-		if mavenHome == "" {
-			return "", errorutils.CheckErrorf("Could not find the location of the maven home directory, by running 'mvn --version' command. The command output is:\n" + stdout.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		if strings.HasPrefix(line, "Apache Maven") {
+			mvnVersion = strings.Split(line, " ")[2]
 		}
 	}
+
+	if mavenHome == "" {
+		return "", errorutils.CheckErrorf("Could not find the location of the maven home directory, by running 'mvn --version' command. The command output is:\n" + strings.Join(output, " ") + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+	}
+
+	if mvnVersion == "" {
+		return "", errorutils.CheckErrorf("Could not parse mvn version by running 'mvn --version' command. The command output is:\n" + strings.Join(output, " ") + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+	}
+
+	err = validateMinimumVersion(mvnVersion)
+	if err != nil {
+		return "", err
+	}
 	log.Debug("Maven home location: ", mavenHome)
+	log.Debug("Maven version: ", mvnVersion)
 	return mavenHome, nil
 }
 
+func runMvnVersionCommand(mavenHome string) ([]string, error) {
+	mvnPath := ""
+	var err error
+	if mavenHome != "" {
+		mvnPath = filepath.Join(mavenHome, "bin", "mvn")
+	} else {
+		mvnPath, err = exec.LookPath("mvn")
+		if err != nil || mvnPath == "" {
+			return nil, errorutils.CheckErrorf(err.Error() + "Hint: The mvn command may not be included in the PATH. Either add it to the path, or set the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		}
+	}
+	cmd := exec.Command(mvnPath, "--version")
+	var stdout bytes.Buffer
+	cmd.Stdout = &stdout
+	err = errorutils.CheckError(cmd.Run())
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.TrimSpace(stdout.String()), "\n"), nil
+}
+
+func parseMvnHome(line string) (string, error) {
+	// The M2_HOME environment variable is not defined.
+	// Since Maven installation can be located in different locations,
+	// Depending on the installation type and the OS (for example: For Mac with brew install: /usr/local/Cellar/maven/{version}/libexec or Ubuntu with debian: /usr/share/maven),
+	// We need to grab the location using the mvn --version command
```

Comment:
It looks like following the code changes done, this description is now misplaced. I think it should be moved to the ```getMavenHomeAndValidateVersion``` function, above the following line - 
```
mavenHome := os.Getenv(MavenHome)
```